# Papers

## fastqc
[Documentation](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/)

Most sequencers will generate a QC report as part of their analysis pipeline, but this is usually only focused on identifying problems which were generated by the sequencer itself. FastQC aims to provide a QC report which can spot problems which originate either in the sequencer or in the starting library material.

## multiqc
[Documentation](https://multiqc.info/docs/)

MultiQC is a reporting tool that parses summary statistics from results and log files generated by other bioinformatics tools. MultiQC doesn't run other tools for you - it's designed to be placed at the end of analysis pipelines or to be run manually when you've finished running your tools.

## samtools
[Documentation](http://www.htslib.org/doc/samtools.html)

Samtools is a set of utilities that manipulate alignments in the BAM format (binary version of SAM). It imports from and exports to the SAM (Sequence Alignment/Map) format, does sorting, merging and indexing, and allows to retrieve reads in any regions swiftly.

## HiSat2
[paper](https://www.nature.com/articles/s41587-019-0201-4)
[manual](http://daehwankimlab.github.io/hisat2/manual/)

HISAT2 is a fast and sensitive alignment program for mapping next-generation sequencing reads (whole-genome, transcriptome, and exome sequencing data) against the general human population (as well as against a single reference genome

## featureCounts
[paper](https://academic.oup.com/bioinformatics/article/30/7/923/232889#2676612)
In module subread.
a software program developed for counting reads to genomic features such as genes, exons, promoters and genomic bins. A read summarization program.
